include/ignition/gui/qml/GzPose.qml

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+import QtQuick 2.9
+import QtQuick.Controls 2.2
+import QtQuick.Controls.Material 2.1
+import QtQuick.Layouts 1.3
+import QtQuick.Controls.Styles 1.4
+
+/**
+ *  Item displaying 3D pose information.
+ *
+ *  Users should load values to xValues, yValues, etc.
+ *  If readOnly == False,
+ *  users can read from signal pararmeters of gzPoseSet: _x, _y, etc.
+ *
+ *  Usage example:
+ *  GzPose {
+ *    id: gzPose
+ *    readOnly: false
+ *    xValue: xValueFromCPP
+ *    yValue: yValueFromCPP
+ *    zValue: zValueFromCPP
+ *    rollValue: rollValueFromCPP
+ *    pitchValue: pitchValueFromCPP
+ *    yawValue: yawValueFromCPP
+ *    onGzPoseSet: {
+ *      myFunc(_x, _y, _z, _roll, _pitch, _yaw)
+ *    }
+ *  }
+**/
+
+Item {
+  id: gzPoseRoot
+
+  // Read-only / write
+  property bool readOnly: false
+
+  // User input value.
+  property double xValue
+  property double yValue
+  property double zValue
+  property double rollValue
+  property double pitchValue
+  property double yawValue
+
+  /**
+   * Used to read spinbox values
+   * @params: _x, _y, _z, _roll, _pitch, _yaw: corresponding spinBoxes values
+   * @note: When readOnly == false, user should read spinbox value from its
+   *        parameters.
+   *        When readOnly == true, this signal is unused.
+   */
+  signal gzPoseSet(double _x, double _y, double _z, double _roll, double _pitch, double _yaw)
+
+
+  /*** The following are private variables: ***/
+  // Show Pose bar (used to control expand)
+  property bool show: true
+
+  height: gzPoseContent.height
+
+  // Left indentation
+  property int indentation: 10
+
+  // Horizontal margins
+  property int margin: 5
+
+  // Maximum spinbox value
+  property double spinMax: 1000000
```

Comment:
I recently had to increase the max value of a pose widget because 100000 was too limiting:

* https://github.com/gazebosim/gz-sim/pull/1221

I think we should use `Number.MAX_VALUE` here.

```suggestion
  property double spinMax: Number.MAX_VALUE
```

---

I'd say this is a public variable. Users of this widget may want to limit the values that can be set. Can we move it up and document it with the public variables?

I realize this opens room for many possibilities - do we want separate limits for position and orientation? Maybe per-spin-box? How about the minimum value? I think these are things to keep in mind, but we don't need to implement any of it until there's a need.